modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/LowStorageDialog.kt

```diff
@@ -0,0 +1,58 @@
+package au.com.shiftyjelly.pocketcasts.settings
+
+import android.content.Context
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsEvent
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsTracker
+import au.com.shiftyjelly.pocketcasts.analytics.SourceView
+import au.com.shiftyjelly.pocketcasts.preferences.Settings
+import au.com.shiftyjelly.pocketcasts.utils.Util
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.Feature
+import au.com.shiftyjelly.pocketcasts.utils.featureflag.FeatureFlag
+import au.com.shiftyjelly.pocketcasts.utils.isDeviceRunningOnLowStorage
+import au.com.shiftyjelly.pocketcasts.views.dialog.ConfirmationDialog
+import au.com.shiftyjelly.pocketcasts.images.R as IR
+import au.com.shiftyjelly.pocketcasts.localization.R as LR
+import au.com.shiftyjelly.pocketcasts.ui.R as UR
+
+class LowStorageDialog(
+    val context: Context,
+    val analyticsTracker: AnalyticsTracker,
+    val settings: Settings,
+) {
+
+    suspend fun shouldShow(downloadedFiles: Long): Boolean = isDeviceRunningOnLowStorage() &&
+        downloadedFiles != 0L &&
+        settings.shouldShowLowStorageModalAfterSnooze() &&
+        FeatureFlag.isEnabled(Feature.MANAGE_DOWNLOADED_EPISODES)
+
+    fun show(
```

Comment:
This function name is a bit awkward. In order to show a dialog you need to do `dialog.show().show()`.